### ISSUE TITLE
Implement std::ops::Add for vectors of field elements

### ISSUE THREAD

**Issue by divergentdave** [divviup/libprio-rs#301]
There are many places in the codebase where we zip or index through two vectors, adding one to the other. We could write blanket implementations of `Add` and `AddAssign` on vectors of field elements to get rid of some boilerplate. These could also check that the lengths of the addend vectors are equal, and panic otherwise to be defensive.

See: https://github.com/divviup/libprio-rs/pull/300#discussion_r966213248

**Comment by cjpatton** [comment]
What's y our idea here @divergentdave? You can't just implement `Add` on `Vec<F>` because `Vec` is a foreign type, i.e., this doesn't compile:

```rust
impl<F: FieldElement> Add for Vec<F> {
    type Output = Vec<F>;

    fn add(self, rhs: Vec<F>) -> Vec<F> {
        todo!()
    }
}
```

We could do this wherever we have a wrapper type, like `OutputShare` or `AggregateShare`. Would that suffice to close this?

**Comment by divergentdave** [comment]
Good point. Implementing it on `OutputShare` and `AggregateShare` sounds good. We could also go through and replace for loops adding field vectors with calls to appropriate private helper functions.

**Comment by cjpatton** [comment]
Alrighty, here's my proposal: https://github.com/divviup/libprio-rs/pull/1205

A couple of caveats:
* I decided to not implement Add and AddAssign for OutputShare and AggregateShare, since the Aggregator API already facilitates arithmetic on these. I suppose we could simplify the implementations of our concrete VDAFs a little, but I didn't bother.
* `vec_assign_vector(a, b)` panics if `a` is longer than `b`, but not if `b` is longer than `a`. The reason is that I wanted to be able to let `b` be an iterator, and there are cases where the iterator has infinite about. For example, in Prio3, `b` is a `Prng`. As far as I can tell, enforcing the length of `b` would require an allocation here, but I could be wrong.
